doc/languages-frameworks/vsc.md

```diff
@@ -0,0 +1,191 @@
+    });
+
+  in
+    programs.vscode = {
+      enable = true;
+      package = finalPackage;
+      extensions = [ ];
+    };
+}
+```
+
+It's important to install all extensions through `vscode-with-extensions` and leave the `extensions` option empty.
+
+### Non-NixOS With Home Manager
+
+On other OSes you also have the option of installing all extensions imperatively, through the VSC UI. In this case Nix will not be in charge of any extensions. The advantage is that your using the blessed and official way of installing extensions. If extensions require extra dependencies though, those might either end up installed globally, or you're required to install them manually. The usual downsides of not using Nix.
+
+## Insiders
+
+You can pass an `isInsiders` argument to the [VSC derivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/vscode.nix#L1). This **only changes the name of the resulting command** and doesn't actually install the insiders version. Which version is installed depends entirely on the URL from which VSC is downloaded, most importantly the version included in the URL string.
+
+The version is [hardcoded](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/vscode.nix#L28) in the derivation, but you can supply your own by overriding the version.
+
+Here is a complete example where the version is overridden so that a very recent VSC version is installed, as `code-insiders`. You would have to regularly update the `sha256` attribute though (tools such as `niv` can help with this).
+
+```nix
+{ pkgs, config, ... }:
+let
+  version = "latest";
+
+  latest = (pkgs.vscode.override {
+    isInsiders = true;
+  }).overrideAttrs
+    (_: rec {
+      src = builtins.fetchurl {
+        name = "VSCode_${version}_${plat}.${archive_fmt}";
+        url = "https://vscode-update.azurewebsites.net/${version}/${plat}/insider";
+        sha256 = "16qlgqfpz7pn52dw6r3xav4ly7rpl5lwck94vdpvzgld1ja9rpm0";
+      };
+    });
+
+  marketplace = pkgs.vscode-utils.extensionsFromVscodeMarketplace ([
+    {
+      name = "vscode-theme-onelight";
+      publisher = "akamud";
+      version = "2.2.3";
+      sha256 = "1mzd77sv6lb6kfv5fvdvzggs488q553cf752byrml981ys9r7khz";
+    }
+  ];
+
+  finalPackage = (pkgs.vscode-with-extensions.override {
+    vscode = latest;
+    vscodeExtensions = with pkgs.vscode-extensions; [ ms-vsliveshare.vsliveshare ] ++ marketplace;
+  }).overrideAttrs (old: {
+    inherit (latest) pname version;
+  });
+in
+{
+  programs.vscode = {
+    enable = true;
+    package = finalPackage;
+    extensions = [ ];
+  };
+}
+```
+
+### Extensions Helper
+
+The script at `https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/vscode.nix#L28` makes updating marketplace extensions much more convenient. It takes a list of extensions on STDIN and outputs objects that you can paste into your Nix code.
```

Comment:
Linking lines will likely break in the future.

Reply:
Hm yeah that's a good point I'll take that out